[Content] HashRouter reference page (#626)
solidjs/solid-docs-next: src/routes/solid-router/reference/components/hash-router.mdx

---
title: HashRouter
---

The `HashRouter` is a top level component that manages the routing of your application.

> It's a client side router, utilizing hash-values in the URL, allow your single page application to simulate a multi-page application.

Reviewer: ```suggestion
It is a client side router that uses hash-values in the URL - providing a single-page application a way to replicate the experience of a multi-page application.
```

> The advantage of hash-routing is that the whole application can run from a single HTML file, which can be hosted on a static file server.

Reviewer: ```suggestion
Since hash-routing provides a way for an application to run from a single HTML file, it can be used when hosting on a static file server.
```

> A disadvantage of this approach, compared to the browser-history like the [`Router`](/solid-router/reference/components/router), is that hash-routing isn't SEO friendly.
> This is because most search engines don't index the hash part of the URL, so they only see the index page of your application.

Reviewer: ```suggestion
Compared to a browser-router, such as [`Router`](/solid-router/reference/components/router), is that this approach is not SEO friendly.
Because most search engines do not index the hashed portion of a URL, they are only able to see the index page of your application when using this approach.
```

> The `root` property is used for components that wrap the matched route, which needs access to the router context.
> This would often be the case for a navigation component with [`<A>`](/solid-router/reference/components/a) links.

Reviewer: ```suggestion
The `root` property is used for components that wrap a matched route and require access to the router context, relevant with navigation components that use [`<A>`](/solid-router/reference/components/a) links.
```
